### PR TITLE
Added a flags to enable azure services.

### PR DESCRIPTION
**Pull request by CalvinRodo**
## Please note:
This is a bit janky ideally we would use a special production.js configuration for this, and wouldn't use the default connection in the datastore.js that is mainly meant for local stuff only. This will change soon as we have prod builds working 

## To Test: 

1. Setup a firewall rule in azure to allow you to connect to an Azure Database for PostgreSQL. 
1. Setup a firewall rule in azure to allow you to connect to an Azure Cache for Redis service.
1. Set the FEATURE_AZURE_PG_SSL flag to true 
1. Set the FEATURE_REDIS_SSL flag to true 
1. Download the Cert specified here to your local folder https://docs.microsoft.com/en-us/azure/postgresql/concepts-ssl-connection-security
1. Run the thing
1. ???
1. Profit (or you know it doesn't crash on startup.)

Also adds the latest version of Lodash

### CODE REVIEW

File: config/datastores.js [hunk 1]
```diff
@@ -13,6 +13,17 @@
  * https://sailsjs.com/config/datastores
  */
 
+function generateDialectOptions()  {
+  if (process.env.AZURE_DB_PG_SSL !== 'true'){
+    return {};
+  }
+  return {
+    'ssl': {
+      ca: require('fs').readFileSync('./BaltimoreCyberTrustRoot.crt.pem')
```
Comment (var-kyle) [inline config/datastores.js]: Can you add `rejectUnauthorized: true,` above the ca: line? I noticed there was a message about this missing when testing locally.

**Comment by dsamojlenko** [comment]
See #158 for production builds

**Comment by CalvinRodo** [comment]
I'll update the description later.

File: config/bootstrap.js [hunk 2]
```diff
@@ -11,6 +11,11 @@
 
 module.exports.bootstrap = async function() {
 
+  sails.log.debug('Environment Variables:');
+  _.forIn(process.env,(value, key) => {
+    sails.log.debug(`${key} : ${value}`);
+  });
+
```
Comment (dsamojlenko) [inline config/bootstrap.js]: We may not want to be logging out environment variables ... there may be some secrets in here?

File: config/datastores.js [hunk 3]
```diff
@@ -48,6 +63,10 @@ module.exports.datastores = {
     *                                                                          *
     ***************************************************************************/
     url: process.env.DATABASE_URL || 'postgresql://postgres@db:5432/postgres',
+    options: {
+      logging: 'verbose',
+      dialectOptions: generateDialectOptions(),
+    }
```
Comment (dsamojlenko) [inline config/datastores.js]: Just noting that we should come back once we are able to run production builds and remove this conditional stuff and move it into config/env/production.js

Comment (dsamojlenko) [reply config/datastores.js]: Maybe put a @TODO comment block above?

File: config/session.js [hunk 4]
```diff
@@ -9,6 +9,14 @@
...
  */
 
+const featureFlags = require('../api/utils/FeatureFlags');
+function generateTls()  {
+  if (featureFlags.isEnabled('FEATURE_REDIS_SSL')) {
+    return { };
+  }
+  return undefined;
+}
+
```
Comment (dsamojlenko) [inline config/session.js]: Same comment as above about coming back around to this later. Maybe add a @TODO comment block.

File: config/bootstrap.js (hunk 2 above)
Comment (dsamojlenko) [reply config/bootstrap.js]: Maybe it's ok because it's only done in debug mode .... I could be convinced.

Comment (var-kyle) [reply config/bootstrap.js]: I agree, not great to have this - removed.

File: config/datastores.js (hunk 3 above)
Comment (var-kyle) [reply config/datastores.js]: Added a todo, but this will likely get resolved in #158

File: config/session.js (hunk 4 above)
Comment (var-kyle) [reply config/session.js]: Same as above, this will get changed in #158

**Pull request merged**